client/src/views/TrackViewer/Layers.vue

```diff
@@ -202,14 +202,8 @@ export default defineComponent({
 
     editAnnotationLayer.$on('update:geojson',
       (data: GeoJSON.Feature<GeoJSON.Polygon>) => {
-        const track = props.trackMap.get(props.selectedTrackId.value);
-        if (track) {
-          const bounds = geojsonToBound(data);
-          track.setFeature({
-            frame: frameNumber.value,
-            bounds,
-          });
-        }
+        const bounds = geojsonToBound(data);
+        emit('update-rect-bounds', frameNumber.value, bounds);
```

Comment:
(3) If this was reverted such that `track.setFeature` is called here, then afterward, the component emitted an `after-track-update` type event, it could be handled in viewer however the current state permits.